code-challenges.test.js

```diff
@@ -23,29 +51,165 @@ const colors2 = [
   "periwinkle",
   "ochre",
   "aquamarine",
-  "saffron"
+  "saffron",
 ]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 // b) Create the function that makes the test pass.
+//Pseudocode:
+// input:An array
+// output: An arrray with the first value removed and the rest of the values shuffled.
+// create a function arrayContaining that takes in an array as a parameter.
+//declare a variable newArray to store the array without the first element or zero indexed value by using the .slice method and passing an argument of one to cut out the string at the zero index.
+//I had to look up how to shuffle remaming array elements in random order. I used the fisher yates shuffle algorithm to accomplish this. By running a for loop that lets the index be the length od the array so it starts from the end of the array and iterates through the array down to the zero index in decrements of 1.
+// declare a variable that randomizes the shuffle of using Math.floor and Math.random methods. Math.random generates a random floating number which we had 1 to. Then that cloat is passed as an argument in Math.floor that rounds down that float to the nearest whole number.
+// by using desctructuring the index and the index of declared variable 'j' are interchanged.
+// return the newArray with the shuffled elements.
+
+// https://www.geeksforgeeks.org/shuffle-a-given-array-using-fisher-yates-shuffle-algorithm/
+const arrayContaining = (array) => {
+  const newArray = array.slice(1)
+  for (let i = newArray.length - 1; i >= 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1))
+    ;[newArray[i], newArray[j]] = [newArray[j], newArray[i]]
+  }
+  return newArray
+}
+
```

Comment:
Nice job researching and implementing the fisher yates shuffle algo! However, the test will fail as it is because you are telling it to expect the function to NOT return an array containing the first item removed and the rest shuffled, when that's in fact what the function is returning!